fs/fs_zenfs.cc

```diff
@@ -379,24 +379,24 @@ IOStatus ZenFS::SyncFileMetadata(ZoneFile* zoneFile) {
 }
 
 /* Must hold files_mtx_ */
-ZoneFile* ZenFS::GetFileInternal(std::string fname) {
-  ZoneFile* zoneFile = nullptr;
+std::shared_ptr<ZoneFile> ZenFS::GetFileInternal(std::string fname) {
+  std::shared_ptr<ZoneFile> zoneFile(nullptr);
```

Comment:
Why do we need a separate method for this?

And the whole point of the get is to increase the ref count, right? We need to return zoneFile.get()

Reply:
It was done to avoid race condition in https://github.com/westerndigitalcorporation/zenfs/pull/25